sentry-rails/lib/sentry/rails/tracing/abstract_subscriber.rb

```diff
@@ -41,14 +41,18 @@ def record_on_current_span(duration:, **options)
             return unless options[:start_timestamp]
 
             scope = Sentry.get_current_scope
-            transaction = scope.get_transaction
-            return unless transaction && transaction.sampled
+            current_span = scope.get_span
+            return unless current_span && current_span.sampled
+
+            span = current_span.start_child(**options)
```

Comment:
We can use `with_child_span(&block)` here:

https://github.com/getsentry/sentry-ruby/blob/341b66e5f3af6643cdb9458611fdacff8f60584d/sentry-ruby/lib/sentry/span.rb#L151-L166